server/routes/money.js

```diff
@@ -15,14 +21,22 @@ function isValidDateSelection(selectedDate, dateSelection) {
 }
 
 function processSelectedDate(selectedDate) {
-  const dateSelection = getDateSelection(new Date(), parseISO(selectedDate));
+  const parsedDate = parseISO(selectedDate);
```

Comment:
I have no idea why this suddenly started to fail. The failing tests were calculating incorrect dates due to trying to use today's date with previous years causing a weird set of results. The underlying function `getDateSelection` is fine and allows for dependency injection but this cannot happen here due to it being a route that's being tested.

I have pulled the current year out of the selectedDate (if parsable) to make `new Date()` behave as if it's the year from the selectedDate rather than when the tests are run.